### PR TITLE
Use or migrate to tectonic-cluo-operator

### PR DESCRIPTION
**Pull request by dghubble**
Deploy Tectonic clusters with `quay.io/coreos/tectonic-cluo-operator:v0.1.3`

* `tectonic-cluo-operator` will manage the CLUO v0.3.1 update-operator (coordinator) deployment and update-agent daemonset
* `update-operator` will stop managing the update-agent daemonset
* `update-operator` will auto-label Container Linux nodes
* `update-agent` will node select for the CLUO `agent=true` label
* Switch AppVersion from TPR to CRD

rel: #1644

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -25,43 +25,43 @@ variable "tectonic_container_images" {
...
 
   default = {
-    addon_resizer                   = "gcr.io/google_containers/addon-resizer:2.0"
-    alertmanager                    = "quay.io/prometheus/alertmanager:v0.7.1"
-    awscli                          = "quay.io/coreos/awscli:025a357f05242fdad6a81e8a6b520098aa65a600"
-    bootkube                        = "quay.io/coreos/bootkube:v0.5.0"
-    calico                          = "quay.io/calico/node:v2.4.1"
-    calico_cni                      = "quay.io/calico/cni:v1.10.0"
-    config_reload                   = "quay.io/coreos/configmap-reload:v0.0.1"
-    container_linux_update_operator = "quay.io/coreos/container-linux-update-operator:v0.2.2"
```
Comment (dghubble) [inline config.tf]: Unfortunately, container_linux_update_operator was the longest name so terraform fmt reformats the whole section and Github shows it as a bunch of changes.

File: modules/tectonic/assets.tf [hunk 2]
```diff
@@ -9,29 +9,29 @@ resource "template_dir" "tectonic" {
...
+    tectonic_monitoring_auth_image     = "${var.container_images["tectonic_monitoring_auth"]}"
+    prometheus_image                   = "${var.container_images["prometheus"]}"
+    prometheus_config_reload_image     = "${var.container_images["prometheus_config_reload"]}"
+    alertmanager_image                 = "${var.container_images["alertmanager"]}"
+    stats_emitter_image                = "${var.container_images["stats_emitter"]}"
+    stats_extender_image               = "${var.container_images["stats_extender"]}"
+    tectonic_channel_operator_image    = "${var.container_images["tectonic_channel_operator"]}"
+    tectonic_prometheus_operator_image = "${var.container_images["tectonic_prometheus_operator"]}"
+    tectonic_etcd_operator_image       = "${var.container_images["tectonic_etcd_operator"]}"
+    tectonic_cluo_operator_image       = "${var.container_images["tectonic_cluo_operator"]}"
```
Comment (dghubble) [inline modules/tectonic/assets.tf]: this is the addition

**Comment by dghubble** [comment]
test upgrade from 1.6.7-tectonic.2 1.6.7-tectonic.2

**Comment by dghubble** [comment]
Manually testing new cluster creation - tectonic-cluo-operator and CLUO look good :+1:

File: modules/update-payload/make-update-payload.sh [hunk 3]
```diff
@@ -100,7 +100,7 @@ for f in ${ASSETS_DIR}/app_versions/*.yaml; do
   # shellcheck disable=SC2086
   metadata=$(yaml2json < ${f} | jq .metadata)
   # shellcheck disable=SC2086
-  desiredVersion=$(yaml2json < ${f} | jq .status.currentVersion)
+  desiredVersion=$(yaml2json < ${f} | jq .spec.desiredVersion)
```
Comment (yifan-gu) [inline modules/update-payload/make-update-payload.sh]: @dghubble Can you change line 84 as well to make it consistent? Thank you!

**Comment by yifan-gu** [comment]
@dghubble Need a rebase.

File: modules/update-payload/make-update-payload.sh (hunk 3 above)
Comment (dghubble) [inline modules/update-payload/make-update-payload.sh]: Using desiredVersion in both places now

**Comment by dghubble** [comment]
Created a 1.7.1-tectonic.1 cluster (old), staging payload, and successfully upgraded to 1.7.3-tectonic.1 (example upgrade). tectonic-cluo-operator migrated the container-linux-update-operator Deployment and container-linux-update-agent DaemonSet.

```
I0822 00:17:46.115190       1 main.go:56] started leading: running tectonic-cluo-operator
I0822 00:17:47.415218       1 update.go:45] Starting Update to target: 0.1.3
I0822 00:17:49.014896       1 update.go:98] Updating component DaemonSet/tectonic-system/container-linux-update-agent-ds
I0822 00:17:49.948159       1 update.go:120] Finished update of component: DaemonSet/tectonic-system/container-linux-update-agent-ds
I0822 00:17:51.116422       1 update.go:98] Updating component Deployment/tectonic-system/container-linux-update-operator
I0822 00:17:51.643543       1 update.go:120] Finished update of component: Deployment/tectonic-system/container-linux-update-operator
I0822 00:17:51.754272       1 update.go:87] Update finished
```

```
I0822 00:18:04.891809       1 main.go:77] /bin/update-operator running
I0822 00:18:05.190629       1 leaderelection.go:179] attempting to acquire leader lease...
I0822 00:20:08.276330       1 leaderelection.go:189] successfully acquired lease tectonic-system/container-linux-update-operator-lock
I0822 00:20:08.373761       1 agent_manager.go:64] Setting label 'agent=true' on "node1.example.com"
I0822 00:20:08.610758       1 agent_manager.go:64] Setting label 'agent=true' on "node2.example.com"
I0822 00:20:08.678072       1 agent_manager.go:64] Setting label 'agent=true' on "node3.example.com"
I0822 00:20:09.611405       1 operator.go:517] Found 0 rebooted nodes
I0822 00:20:40.539349       1 operator.go:517] Found 0 rebooted nodes
```

**Comment by Quentin-M** [comment]
WOOOOOWW Coooool !

**Comment by Quentin-M** [comment]
Dalton FTW ~

**Comment by dghubble** [comment]
Here's the current state on the crd_migration branch with a simulated update.

screenshot from 2017-08-21 17-37-38

**Pull request merged**